src/setting_sbp_cb.c

```diff
@@ -303,7 +303,7 @@ static void setting_read_by_index_done_callback(uint16_t sender_id, uint8_t len,
 
   /* Traverse the pending requests */
   request_state_t *state = ctx->req_list;
-  while (state != NULL) {
+  while (state != NULL && state->msg_id == SBP_MSG_SETTINGS_READ_BY_INDEX_REQ) {
```

Comment:
I think you should leave the while loop alone and add an if (state->msg_id == SBP_MSG_SETTINGS_READ_BY_INDEX_REQ) for request_state_signal.

We still need the while loop to call state = state->next;

Reply:
That will consume all the remaining states. Is that expected?

Reply:
I am not 100% sure. I do not know the original intention of this code block, but from the comments it seems like that is the intention, to /* Traverse the pending requests */.